### ISSUE TITLE
In PSTR the IPR scenarios for landuse should not be NA

### ISSUE THREAD

**Issue by maurolepore** [2DegreesInvesting/tiltIndicator#315]
@AnneSchoenauer [said](https://2investinginitiative.slack.com/archives/C050VAQACC9/p1683910479957129?thread_ts=1683893746.777999&cid=C050VAQACC9)

> If you look at this sheet https://docs.google.com/spreadsheets/d/1ViTkhidRxcMLqlakH9oN4POT2RvTP0mCDm2yoi1TCU4/edit you see that there are NAs for the land use sector. This shouldn’t be the case as the land use sector should be covered by the IPR scenarios. 

https://github.com/2DegreesInvesting/tiltIndicator/issues/312#issuecomment-1546033719

@AnneSchoenauer also said 

> Dear @Tilmon and dear @maurolepore, 
I randomly checked now some companies and I noticed one issue which I however don't know if it is on purpose, if it is a technicality or if is content-related. Therefore, @maurolepore if you don't see that it is a code problem, I am afraid that we need to wait for @Tilmon to tell us what he thinks about the issue on Monday. 
I noticed that some of the companies don't have any sector and therefore land in the risk category "no_sector". However in this sample company [here](https://docs.google.com/spreadsheets/d/1ViTkhidRxcMLqlakH9oN4POT2RvTP0mCDm2yoi1TCU4/edit#gid=975827873), we should have sectors namely the "Land Use" sector that should be matched with a sector reduction target of IPR. I didn't check if now all "Land Use" sectors are not matched to a scenario - so I don't know if the sample company is an excemption but this is, so far as I know, not correct. 
Other than that the outputs are great and look exactly of how we would like to see them. 
@maurolepore if you are nearly there with running the whole PSTR I would suggest that you continue to finish the outputs. So that if this issue here is not a real issue, @Tilmon can work with the output tables on Monday. 

https://github.com/2DegreesInvesting/tiltIndicator/issues/316#issue-1708280334

**Comment by maurolepore** [comment]
``` r
library(dplyr, warn.conflicts = FALSE)
library(tiltIndicator)
packageVersion("tiltIndicator")
#> [1] '0.0.0.9058'

# Basic datasets
companies <- tibble(
  type = "x",
  sector = "x",
  subsector = "x",
  # Irrelevant
  company_id = "x",
  clustered = "x",
  activity_uuid_product_uuid = "x",
  isic_4digit = "x",
  tilt_sector = "x",
  tilt_subsector = c("x", "x"),
)
scenarios <- tibble(
  type = "x",
  sector = "x",
  subsector = "x",
  # Irrelevant
  scenario = "x",
  year = 1,
  value = 1,
  reductions = 1:2,
)

# `pstr*()` match companies to scenarios by all of type, sector, and subsector
# Here they are all the same so they match
companies <- companies |> mutate(sector = c("land use", "industry"))
scenarios <- scenarios |> mutate(sector = c("land use", "industry"))
companies
#> # A tibble: 2 × 9
#>   type  sector subsector company_id clustered activity_uuid_produc…¹ isic_4digit
#>   <chr> <chr>  <chr>     <chr>      <chr>     <chr>                  <chr>      
#> 1 x     land … x         x          x         x                      x          
#> 2 x     indus… x         x          x         x                      x          
#> # ℹ abbreviated name: ¹​activity_uuid_product_uuid
#> # ℹ 2 more variables: tilt_sector <chr>, tilt_subsector <chr>
scenarios
#> # A tibble: 2 × 7
#>   type  sector   subsector scenario  year value reductions
#>   <chr> <chr>    <chr>     <chr>    <dbl> <dbl>      <int>
#> 1 x     land use x         x            1     1          1
#> 2 x     industry x         x            1     1          2
pstr_at_product_level(companies, scenarios)
#> # A tibble: 2 × 10
#>   companies_id grouped_by risk_category clustered activity_uuid_product_uuid
#>   <chr>        <chr>      <chr>         <chr>     <chr>                     
#> 1 x            x_x_1      low           x         x                         
#> 2 x            x_x_1      low           x         x                         
#> # ℹ 5 more variables: tilt_sector <chr>, tilt_subsector <chr>, scenario <chr>,
#> #   year <dbl>, type <chr>
pstr(companies, scenarios)
#> # A tibble: 3 × 4
#>   companies_id grouped_by risk_category value
#>   <chr>        <chr>      <chr>         <dbl>
#> 1 x            x_x_1      high              0
#> 2 x            x_x_1      medium            0
#> 3 x            x_x_1      low               1

# Here they are NOT all the same so some don't match
companies <- companies |> mutate(sector = c("land use", "industry"))
scenarios <- scenarios |> mutate(sector = c("land use", "other"))
companies
#> # A tibble: 2 × 9
#>   type  sector subsector company_id clustered activity_uuid_produc…¹ isic_4digit
#>   <chr> <chr>  <chr>     <chr>      <chr>     <chr>                  <chr>      
#> 1 x     land … x         x          x         x                      x          
#> 2 x     indus… x         x          x         x                      x          
#> # ℹ abbreviated name: ¹​activity_uuid_product_uuid
#> # ℹ 2 more variables: tilt_sector <chr>, tilt_subsector <chr>
scenarios
#> # A tibble: 2 × 7
#>   type  sector   subsector scenario  year value reductions
#>   <chr> <chr>    <chr>     <chr>    <dbl> <dbl>      <int>
#> 1 x     land use x         x            1     1          1
#> 2 x     other    x         x            1     1          2
pstr_at_product_level(companies, scenarios)
#> # A tibble: 2 × 10
#>   companies_id grouped_by risk_category clustered activity_uuid_product_uuid
#>   <chr>        <chr>      <chr>         <chr>     <chr>                     
#> 1 x            x_x_1      low           x         x                         
#> 2 x            x_NA_NA    no_sector     x         x                         
#> # ℹ 5 more variables: tilt_sector <chr>, tilt_subsector <chr>, scenario <chr>,
#> #   year <dbl>, type <chr>
pstr(companies, scenarios)
#> # A tibble: 6 × 4
#>   companies_id grouped_by risk_category value
#>   <chr>        <chr>      <chr>         <dbl>
#> 1 x            x_NA_NA    high             NA
#> 2 x            x_NA_NA    medium           NA
#> 3 x            x_NA_NA    low              NA
#> 4 x            x_x_1      high              0
#> 5 x            x_x_1      medium            0
#> 6 x            x_x_1      low               1

# Same here
companies <- companies |> mutate(sector = c("land use", "industry"))
scenarios <- scenarios |> mutate(sector = c("land use", NA))
companies
#> # A tibble: 2 × 9
#>   type  sector subsector company_id clustered activity_uuid_produc…¹ isic_4digit
#>   <chr> <chr>  <chr>     <chr>      <chr>     <chr>                  <chr>      
#> 1 x     land … x         x          x         x                      x          
#> 2 x     indus… x         x          x         x                      x          
#> # ℹ abbreviated name: ¹​activity_uuid_product_uuid
#> # ℹ 2 more variables: tilt_sector <chr>, tilt_subsector <chr>
scenarios
#> # A tibble: 2 × 7
#>   type  sector   subsector scenario  year value reductions
#>   <chr> <chr>    <chr>     <chr>    <dbl> <dbl>      <int>
#> 1 x     land use x         x            1     1          1
#> 2 x     <NA>     x         x            1     1          2
pstr_at_product_level(companies, scenarios)
#> # A tibble: 2 × 10
#>   companies_id grouped_by risk_category clustered activity_uuid_product_uuid
#>   <chr>        <chr>      <chr>         <chr>     <chr>                     
#> 1 x            x_x_1      low           x         x                         
#> 2 x            x_NA_NA    no_sector     x         x                         
#> # ℹ 5 more variables: tilt_sector <chr>, tilt_subsector <chr>, scenario <chr>,
#> #   year <dbl>, type <chr>
pstr(companies, scenarios)
#> # A tibble: 6 × 4
#>   companies_id grouped_by risk_category value
#>   <chr>        <chr>      <chr>         <dbl>
#> 1 x            x_NA_NA    high             NA
#> 2 x            x_NA_NA    medium           NA
#> 3 x            x_NA_NA    low              NA
#> 4 x            x_x_1      high              0
#> 5 x            x_x_1      medium            0
#> 6 x            x_x_1      low               1
```

<sup>Created on 2023-05-12 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>




```r
tibble::tribble(
  ~type,    ~sector,       ~subsector,       ~tilt_sector,                                   ~tilt_subsector,
  "ipr", "land use",               NA,         "land use",                                                NA,
  "ipr", "land use",               NA,         "land use",                                                NA,
  "ipr", "land use",               NA,         "land use",                                                NA,
  "ipr", "industry", "other industry", "steel and metals",                                    "other metals",
  "ipr",         NA,               NA,         "land use", "fishing and forestry; agriculture and livestock"
  )
```

**Comment by maurolepore** [comment]
I think this reprex captures this issue. 

The `pstr*()` functions look for a match in all three columns: `type`, `sector`, and `subsector`. If the match is not perfect, some rows get `NA`s. The code behind this behaviour a `left_join()` between `companies` and `scenarios` by all those columns. 

What behaviour would you expect? 

``` r
library(dplyr, warn.conflicts = FALSE)
library(tiltIndicator)
packageVersion("tiltIndicator")
#> [1] '0.0.0.9058'

# Basic datasets
companies <- tibble(
  type = "x",
  sector = "x",
  subsector = "x",
  # Irrelevant
  company_id = "x",
  clustered = "x",
  activity_uuid_product_uuid = "x",
  isic_4digit = "x",
  tilt_sector = "x",
  tilt_subsector = c("x", "x"),
)
scenarios <- tibble(
  type = "x",
  sector = "x",
  subsector = "x",
  # Irrelevant
  scenario = "x",
  year = 1,
  value = 1,
  reductions = 1:2,
)

# A bit more realistic

# `pstr*()` match companies to scenarios by all of type, sector, and subsector
# Here they are all the same so they match
companies <- companies |> mutate(sector = c("land use", "industry"))
scenarios <- scenarios |> mutate(sector = c("land use", "industry"))
companies
#> # A tibble: 2 × 9
#>   type  sector subsector company_id clustered activity_uuid_produc…¹ isic_4digit
#>   <chr> <chr>  <chr>     <chr>      <chr>     <chr>                  <chr>      
#> 1 x     land … x         x          x         x                      x          
#> 2 x     indus… x         x          x         x                      x          
#> # ℹ abbreviated name: ¹​activity_uuid_product_uuid
#> # ℹ 2 more variables: tilt_sector <chr>, tilt_subsector <chr>
scenarios
#> # A tibble: 2 × 7
#>   type  sector   subsector scenario  year value reductions
#>   <chr> <chr>    <chr>     <chr>    <dbl> <dbl>      <int>
#> 1 x     land use x         x            1     1          1
#> 2 x     industry x         x            1     1          2
pstr_at_product_level(companies, scenarios)
#> # A tibble: 2 × 10
#>   companies_id grouped_by risk_category clustered activity_uuid_product_uuid
#>   <chr>        <chr>      <chr>         <chr>     <chr>                     
#> 1 x            x_x_1      low           x         x                         
#> 2 x            x_x_1      low           x         x                         
#> # ℹ 5 more variables: tilt_sector <chr>, tilt_subsector <chr>, scenario <chr>,
#> #   year <dbl>, type <chr>
pstr(companies, scenarios)
#> # A tibble: 3 × 4
#>   companies_id grouped_by risk_category value
#>   <chr>        <chr>      <chr>         <dbl>
#> 1 x            x_x_1      high              0
#> 2 x            x_x_1      medium            0
#> 3 x            x_x_1      low               1

# Here they are NOT all the same so some don't match
companies <- companies |> mutate(sector = c("land use", "industry"))
scenarios <- scenarios |> mutate(sector = c("land use", "other"))
companies
#> # A tibble: 2 × 9
#>   type  sector subsector company_id clustered activity_uuid_produc…¹ isic_4digit
#>   <chr> <chr>  <chr>     <chr>      <chr>     <chr>                  <chr>      
#> 1 x     land … x         x          x         x                      x          
#> 2 x     indus… x         x          x         x                      x          
#> # ℹ abbreviated name: ¹​activity_uuid_product_uuid
#> # ℹ 2 more variables: tilt_sector <chr>, tilt_subsector <chr>
scenarios
#> # A tibble: 2 × 7
#>   type  sector   subsector scenario  year value reductions
#>   <chr> <chr>    <chr>     <chr>    <dbl> <dbl>      <int>
#> 1 x     land use x         x            1     1          1
#> 2 x     other    x         x            1     1          2
pstr_at_product_level(companies, scenarios)
#> # A tibble: 2 × 10
#>   companies_id grouped_by risk_category clustered activity_uuid_product_uuid
#>   <chr>        <chr>      <chr>         <chr>     <chr>                     
#> 1 x            x_x_1      low           x         x                         
#> 2 x            x_NA_NA    no_sector     x         x                         
#> # ℹ 5 more variables: tilt_sector <chr>, tilt_subsector <chr>, scenario <chr>,
#> #   year <dbl>, type <chr>
pstr(companies, scenarios)
#> # A tibble: 6 × 4
#>   companies_id grouped_by risk_category value
#>   <chr>        <chr>      <chr>         <dbl>
#> 1 x            x_NA_NA    high             NA
#> 2 x            x_NA_NA    medium           NA
#> 3 x            x_NA_NA    low              NA
#> 4 x            x_x_1      high              0
#> 5 x            x_x_1      medium            0
#> 6 x            x_x_1      low               1

# Same here
companies <- companies |> mutate(sector = c("land use", "industry"))
scenarios <- scenarios |> mutate(sector = c("land use", NA))
companies
#> # A tibble: 2 × 9
#>   type  sector subsector company_id clustered activity_uuid_produc…¹ isic_4digit
#>   <chr> <chr>  <chr>     <chr>      <chr>     <chr>                  <chr>      
#> 1 x     land … x         x          x         x                      x          
#> 2 x     indus… x         x          x         x                      x          
#> # ℹ abbreviated name: ¹​activity_uuid_product_uuid
#> # ℹ 2 more variables: tilt_sector <chr>, tilt_subsector <chr>
scenarios
#> # A tibble: 2 × 7
#>   type  sector   subsector scenario  year value reductions
#>   <chr> <chr>    <chr>     <chr>    <dbl> <dbl>      <int>
#> 1 x     land use x         x            1     1          1
#> 2 x     <NA>     x         x            1     1          2
pstr_at_product_level(companies, scenarios)
#> # A tibble: 2 × 10
#>   companies_id grouped_by risk_category clustered activity_uuid_product_uuid
#>   <chr>        <chr>      <chr>         <chr>     <chr>                     
#> 1 x            x_x_1      low           x         x                         
#> 2 x            x_NA_NA    no_sector     x         x                         
#> # ℹ 5 more variables: tilt_sector <chr>, tilt_subsector <chr>, scenario <chr>,
#> #   year <dbl>, type <chr>
pstr(companies, scenarios)
#> # A tibble: 6 × 4
#>   companies_id grouped_by risk_category value
#>   <chr>        <chr>      <chr>         <dbl>
#> 1 x            x_NA_NA    high             NA
#> 2 x            x_NA_NA    medium           NA
#> 3 x            x_NA_NA    low              NA
#> 4 x            x_x_1      high              0
#> 5 x            x_x_1      medium            0
#> 6 x            x_x_1      low               1
```

<sup>Created on 2023-05-12 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by AnneSchoenauer** [comment]
Dear @maurolepore thanks a lot for this! This really helps and is as I can judge a content problem - I will talk to @Tilmon tomorrow and ask him why this is the case and also investigate myself again the scenario mapper - I think at the end the scenario mapper might be wrong. Thanks again! 
All the best 
Anne

**Comment by Tilmon** [comment]
@maurolepore @kalashsinghal @AnneSchoenauer Issue is resolved and sectors updated in new data. Shared new data with Kalash in this ticket [Adapt str_companies & re-run PSTR after Tilman fixed scenario sectors#356](https://github.com/2DegreesInvesting/tiltIndicator/issues/356)
